### PR TITLE
Tron - TG types

### PR DESCRIPTION
**Pull request by phoorichet**
- Add TRX and TRC20 type for Tron TG integration
- Add TRX Solidity bin and ABI for client calls
- Modify fake context to lookup registry by name

### CODE REVIEW

File: auth/tron-signer.go [hunk 1]
```diff
@@ -5,16 +5,34 @@ package auth
...
+	}
+
+	privKey, _ := btcec.PrivKeyFromBytes(btcec.S256(), common.FromHex(string(privateKey)))
+	return &TronSigner{PrivateKey: privKey.ToECDSA()}
+}
+
 func (k *TronSigner) Sign(txBytes []byte) []byte {
-	signature, err := crypto.Sign(sha3.SoliditySHA3(txBytes), k.PrivateKey)
+	signature, err := evmcompat.GenerateTypedSig(
+		sha3.SoliditySHA3(txBytes), k.PrivateKey, evmcompat.SignatureType_EIP712,
```
Comment (enlight) [inline auth/tron-signer.go]: So in loom-js @tstxni97 used `3` not `0` for the sig type https://github.com/loomnetwork/loom-js/pull/268/files#diff-d53e0b623e5d95fd76ae5f1121c4baabR31 aka `SignatureType_TRON`

File: client/gateway/dappchain_gateway.go [hunk 2]
```diff
@@ -93,6 +108,39 @@ func (tg *DAppChainGateway) AddContractMapping(from common.Address, to loom.Addr
...
+		return err
+	}
+	fmt.Printf("Mapping contract %v to %v\n", fromAddr, to)
+
+	hash := ssha.SoliditySHA3(
+		ssha.Address(from),
+		ssha.Address(common.BytesToAddress(to.Local)),
+	)
+
+	sig, err := evmcompat.GenerateTypedSig(hash, creator.MainnetPrivKey, evmcompat.SignatureType_EIP712)
```
Comment (enlight) [inline client/gateway/dappchain_gateway.go]: Same sig type issue here that I've mentioned above

File: client/gateway/dappchain_gateway.go [hunk 3]
```diff
@@ -93,6 +108,39 @@ func (tg *DAppChainGateway) AddContractMapping(from common.Address, to loom.Addr
 	return err
 }
 
+// AddTronContractMapping same as AddContractMapping but for Tron
+func (tg *DAppChainGateway) AddTronContractMapping(from common.Address, to loom.Address,
+	creator *client.Identity, contractTxHash string) error {
```
Comment (enlight) [inline client/gateway/dappchain_gateway.go]: This doesn't need the `contractTxHash` anymore does it?

File: crypto/crypto.go [hunk 4]
```diff
@@ -25,3 +31,32 @@ func LoadEd25519PrivKey(path string) ([]byte, error) {
...
+	buf := make([]byte, 64)
+	if _, err := io.ReadFull(fd, buf); err != nil {
+		return nil, err
+	}
+	privKey, _ := btcec.PrivKeyFromBytes(btcec.S256(), common.FromHex(string(buf)))
+	return privKey.ToECDSA(), nil
+}
+
+// LoadBtecSecp256k1PrivKeyByte reads 64 byte from private key file
+func LoadBtecSecp256k1PrivKeyByte(file string) ([]byte, error) {
```
Comment (enlight) [inline crypto/crypto.go]: This doesn't appear to be used anywhere

File: auth/tron-signer.go [hunk 5]
```diff
@@ -5,16 +5,34 @@ package auth
 import (
 	"crypto/ecdsa"
 
+	"github.com/btcsuite/btcd/btcec"
```
Comment (enlight) [inline auth/tron-signer.go]: Ugh... so go-loom is using `github.com/btcsuite/btcd/btcec` and loomchain is using `github.com/tendermint/btcd/btcec`? That doesn't seem like a good idea, should just use `github.com/btcsuite/btcd/btcec` everywhere I suppose, in which case it shouldn't be vendored, need to pull a specific revision in the Makefile like we do with the Hashicorp plugin & go-ethereum.

File: auth/tron-signer.go (hunk 1 above)
Comment (phoorichet) [reply auth/tron-signer.go]: fixed

File: client/gateway/dappchain_gateway.go (hunk 2 above)
Comment (phoorichet) [reply client/gateway/dappchain_gateway.go]: fixed

File: client/gateway/dappchain_gateway.go (hunk 3 above)
Comment (phoorichet) [reply client/gateway/dappchain_gateway.go]: not anymore

File: crypto/crypto.go [hunk 6]
```diff
@@ -25,3 +31,27 @@ func LoadEd25519PrivKey(path string) ([]byte, error) {
 
 	return privKey, nil
 }
+
+// LoadBtecSecp256k1PrivKey converts private key from btec secp256k1 to ecdsa
+func LoadBtecSecp256k1PrivKey(file string) (*ecdsa.PrivateKey, error) {
+	buf, err := LoadBtecSecp256k1PrivKeyByte(file)
+	if err != nil {
+		return nil, err
```
Comment (enlight) [inline crypto/crypto.go]: -> `return nil, errors.Wrap(err, "failed to load btec secp256k1 private key")`

**Pull request merged**